### ISSUE TITLE
Tools: Implement "Dependency" or "Links" view (e.g. in Loader)

### ISSUE THREAD

**Issue by BigRoy** [ynput/ayon-core#780]
**Is your feature request related to a problem? Please describe.**

> The "Dependency" loader will get more useful after ynput/OpenPype#3629 and ynput/OpenPype#3633. However, to easily trace from a selected version I'd recommend implementing some additional changes.

> The current state:
![afbeelding](https://user-images.githubusercontent.com/2439881/183627949-5d598519-c7e7-467c-9524-df60b437efde.png)

**The above was the state in OpenPype** - In AYON this is currently gone since a rewrite occurred of the Loader tool. This issue is to track the request to add it again, and preferably in a better fashion than before.

**Describe the solution you'd like**

- Allow to identify the Link type, e.g. color code, icon or tooltip based on whether the link is "generative" or "reference"
- Allow to double click to 'go to' in the Loader. So that your asset+subset+version selection in the Loader changes to that Input/Output.
- Potentially even "visualize" whether that 'input version' used is the latest version of that subset. So that you could select a PUBLISH and see from its Inputs that it's not publish using the latest input version.

**Describe alternatives you've considered**

Of course better tooling to visualize the dependencies in general would be even better, like a node graph (e.g. [ReactFlow](https://reactflow.dev/)) or even just showing the recursive inputs/outputs in a hierarchy would be useful, like [this example of what we had used in one of our productions before OpenPype](https://github.com/getavalon/core/issues/395#issuecomment-778078006).

---

It's good to be aware that on the web frontend there's a similar feature in the ["Node Graph" addon](https://ynput.io/ayon/addons/node-graph/) (in beta, low priority backlog) - as Milan mentioned [here](https://discord.com/channels/517362899170230292/890894521817260052/1262801083559051407)

[cuID:[OP-3715](https://app.clickup.com/t/2pkvzqz)]

**Comment by BigRoy** [comment]
Came up on Discord [here](https://discord.com/channels/517362899170230292/890894521817260052/1262877734636355746):

I commented these 'reference' links:

> In OpenPype there was the Dependency Links shown in the Loader tool but it's been removed from AYON when rewriting large parts of the UIs (plus the UI sucked a bit in OpenPype). I think it still has merit to have something nearby somewhere.
> 
> Reference of links in loader: https://github.com/ynput/OpenPype/pull/3629#issuecomment-1209153823 + https://github.com/ynput/OpenPype/pull/3629#issuecomment-1243870103
> 
> I had this tiny widget in avalon: https://github.com/BigRoy/core/blob/acre/avalon/tools/representationinput.py which was basically the screenshot shown here: https://github.com/getavalon/core/issues/395#issuecomment-501749753 which we actually ended up using in a project back then, more details here: https://github.com/getavalon/core/issues/395#issuecomment-778078006 with the gist linking to the UI.
> 
> Originating from these Avalon issues at the time: https://github.com/getavalon/core/issues/395
> 
> It wouldn't be *too* hard to transfer these to AYON codebase since the data is relatively similar, just a different query.
> 
> In one of those threads I also had a NodeGraphQt demo, here: https://github.com/getavalon/core/issues/395#issuecomment-518670047

**Comment by mkolar** [comment]
This issue will be resolved on the web frontend and once that is done, we'll use the same view within DCC tools as well. Closing this as it won't be a matter for core
